internal/operator-controller/applier/helm.go

```diff
@@ -209,7 +213,119 @@ func (h *Helm) buildHelmChart(bundleFS fs.FS, ext *ocv1.ClusterExtension) (*char
 	if err != nil {
 		return nil, err
 	}
-	return h.BundleToHelmChartConverter.ToHelmChart(source.FromFS(bundleFS), ext.Spec.Namespace, watchNamespace)
+	var result *chart.Chart
+	contentType := bundleContentType(bundleFS)
+	switch contentType {
+	case HelmContent:
+		if !features.OperatorControllerFeatureGate.Enabled(features.HelmChartSupport) {
+			return nil, fmt.Errorf("helm chart support is not enabled")
```

Comment:
Since this will be an alpha feature gate, we probably want to word this differently--this current sentence makes it sound like helm support could be enabled as a supported feature. Changing it to something like "installing helm charts is not supported" or "cannot install helm chart content type" would still communicate that we properly detected the content as a helm chart without implying that the user can/should toggle support for it.

Reply:
Should the content negotiation code also be wrapped by the feature gate instead of just the execution of the implementation?

Reply:
That is a great point. We might want to wrap the whole code under the feature gate so the `switch contentType {` is just valid if the feature flag is used.

Reply:
The code has been wrapped by the feature gate per the recommendations.